apps/emqx_bridge/src/emqx_bridge_api.erl

```diff
@@ -1164,3 +1164,19 @@ upgrade_type(Type) ->
 
 downgrade_type(Type) ->
     emqx_bridge_lib:downgrade_type(Type).
+
+%% TODO: move it to callback
```

Comment:
Yes, I agree, this should be made a callback before we start refactoring more bridges but could be done in a later PR.